src/converter/one/convert/JfrToOtlp.java

```diff
@@ -0,0 +1,271 @@
+import one.proto.Proto;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.*;
+
+/** Converts .jfr output to OpenTelemetry protocol. */
+public class JfrToOtlp extends JfrConverter {
+    // Size in bytes to be allocated in the buffer to hold the varint containing the length of the message
+    private static final int MSG_LARGE = 5;
+    private static final int MSG_SMALL = 1;
+
+    private final Index<String> stringPool = new Index<>(String.class, "");
+    private final Index<String> functionPool = new Index<>(String.class, "");
+    private final Index<Line> linePool = new Index<>(Line.class, Line.EMPTY);
+    private final Index<KeyValue> attributesPool = new Index<>(KeyValue.class, KeyValue.EMPTY);
+
+    private final Proto proto = new Proto(1024);
+    private final double ticksPerNanosecond;
+
+    public JfrToOtlp(JfrReader jfr, Arguments args) {
+        super(jfr, args);
+        ticksPerNanosecond = jfr.ticksPerSec / 1_000_000_000.0;
+    }
+
+    @Override
+    public void convert() throws IOException {
+        long rpMark = proto.startField(PROFILES_DATA_resource_profiles, MSG_LARGE);
+        long spMark = proto.startField(RESOURCE_PROFILES_scope_profiles, MSG_LARGE);
+        super.convert();
+        proto.commitField(spMark);
+        proto.commitField(rpMark);
+
+        writeProfileDictionary();
+    }
+
+    @Override
+    protected void convertChunk() {
+        long pMark = proto.startField(SCOPE_PROFILES_profiles, MSG_LARGE);
+
+        writeSampleTypes();
+        writeTimingInformation();
+
+        List<Integer> locationIndices = new ArrayList<>();
+        collector.forEach(new OtlpEventToSampleVisitor(locationIndices));
+
+        long liMark = proto.startField(PROFILE_location_indices, MSG_LARGE);
+        locationIndices.forEach(proto::writeInt);
+        proto.commitField(liMark);
+
+        proto.commitField(pMark);
+    }
+
+    private void writeSampleTypes() {
+        long stsMark = proto.startField(PROFILE_sample_type, MSG_SMALL);
+        proto.field(VALUE_TYPE_type_strindex, stringPool.index(getValueType()));
+        proto.field(VALUE_TYPE_unit_strindex, stringPool.index(getSampleUnits()));
+        proto.field(VALUE_TYPE_aggregation_temporality, AGGREGATION_TEMPORARALITY_cumulative);
+        proto.commitField(stsMark);
+
+        long sttMark = proto.startField(PROFILE_sample_type, MSG_SMALL);
+        proto.field(VALUE_TYPE_type_strindex, stringPool.index(getValueType()));
+        proto.field(VALUE_TYPE_unit_strindex, stringPool.index(getTotalUnits()));
+        proto.field(VALUE_TYPE_aggregation_temporality, AGGREGATION_TEMPORARALITY_cumulative);
+        proto.commitField(sttMark);
+    }
+
+    private void writeTimingInformation() {
+        proto.field(PROFILE_time_nanos, jfr.chunkStartNanos);
+        proto.field(PROFILE_duration_nanos, jfr.chunkDurationNanos());
+    }
+
+    public void dump(OutputStream out) throws IOException {
```

Comment:
The method is lost somewhere in the middle between private methods.
Move it next to other public methods, either after constructor or before `convert`.